src/components/Header.jsx

```diff
@@ -0,0 +1,129 @@
+import React, { useMemo } from "react";
+import AppBar from "@mui/material/AppBar";
+import Box from "@mui/material/Box";
+import Toolbar from "@mui/material/Toolbar";
+import Container from "@mui/material/Container";
+import Button from "@mui/material/Button";
+import { useNavigate, useLocation } from "react-router";
+import { IconButton, Menu, MenuItem, Typography } from "@mui/material";
+import MenuIcon from "@mui/icons-material/Menu";
+
+const pages = [
+  { page: "Home", route: "/" },
+  { page: "Elixirs", route: "/elixirs" },
+  { page: "Houses", route: "/houses" },
+  { page: "Ingredients", route: "/ingredients" },
+  { page: "Spells", route: "/spells" },
+  { page: "Wizards", route: "/wizards" },
+];
+
+function Header() {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const currentRoute = location.pathname;
+  const currentPage =
+    useMemo(
+      () => pages.filter((item) => item.route === currentRoute)[0],
+      [currentRoute]
+    ) || pages[0];
+
+  const [selectedButton, setSelectedButton] = React.useState(currentPage.page);
```

Comment:
You could have just used `useEffect` with empty dependency array to make sure that on every first render selectedButton state has current location.

```suggestion
  const [selectedButton, setSelectedButton] = React.useState();
  
  useEffect(() => {
  setSelectedButton(location.pathname);
  }, [])
```


Using useMemo looks a bit complicated.

Reply:
@hannamaskalenka I found this in the documentation, I might not be understanding something correctly, but it seems like not the right place to use the useEffect: https://react.dev/learn/you-might-not-need-an-effect
<img width="916" alt="image" src="https://github.com/user-attachments/assets/08f317e5-34b0-4304-bca6-197ccbc63e77">

Reply:
> @hannamaskalenka I found this in the documentation, I might not be understanding something correctly, but it seems like not the right place to use the useEffect: https://react.dev/learn/you-might-not-need-an-effect <img alt="image" width="916" src="https://private-user-images.githubusercontent.com/56334517/395191899-08f317e5-34b0-4304-bca6-197ccbc63e77.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQwMTc4MzgsIm5iZiI6MTczNDAxNzUzOCwicGF0aCI6Ii81NjMzNDUxNy8zOTUxOTE4OTktMDhmMzE3ZTUtMzRiMC00MzA0LWJjYTYtMTk3Y2NiYzYzZTc3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjEyVDE1MzIxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMwNDE5NGQ2OGI3MTc5YjlhNjI3YThhMWU4MjUzYzQ2M2VjZDZmMTFjZWZhMjE1MDE3NDMxOTE0NGNlZmEwYTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.zkw1Shf2apVnE5VyNz8VXYykfKbTqhEQ7FIZqBkKciU">

yes, you're right, let's leave it like that for now. I have something else in mind to simplify this component 😏